### DISCUSSION TITLE
What's the real usage of `idHint`?

### DISCUSSION THREAD

**Discussion by nyngwang** [webpack/webpack#15645] [Q&A]
From the [webpack v5 migration guide](https://webpack.js.org/migrate/5/#clean-up-configuration), point #12:

> When using a custom configuration, drop name: false and replace name: string | function with idHint: string | function.

It's suggesting me to change from `name` to `idHint`. But the page [`splitChunks.cacheGroups.{cacheGroup}.idHint`](https://webpack.js.org/plugins/split-chunks-plugin/#splitchunkscachegroupscachegroupidhint) shows no information about what's the benefit of doing this?

**Comment by vankop** [comment]
it will be used as `[name]` otherwise as I see `cacheGroups.key` will be used

**Reply by nyngwang** [reply]
@vankop: Then why not just use `name`? I'm already using this:

```js
name: (mod) => path.parse(mod.identifier()).base
```

So it's suggesting me to change this to `idHint: (mod) => ...`? But if there is no difference why should I do so?

**Reply by vankop** [reply]
it looks like ( from the code ) you can have multiple idHints not sure how they are merging..

**Reply by nyngwang** [reply]
And there is no such a thing `cacheGroups.key` in the page: https://webpack.js.org/plugins/split-chunks-plugin/#optimizationsplitchunks. Are we talking about the same thing :/ ? I just did a Ctrl-F to find the `cacheGroups.key` you mentioned.

**Reply by vankop** [reply]
key is
```js
cacheGroups: { default: {} }
```
`default` is a key

**Reply by nyngwang** [reply]
@vankop: OK, but still the same question: If I already have `name` why use `idHint`? Is this a renaming or something?

**Reply by vankop** [reply]
not sure, from defaults I see `idHint` usage...
